cranelift/codegen/src/isa/x64/abi.rs

```diff
@@ -352,13 +352,14 @@ impl ABIMachineSpec for X64ABIMachineSpec {
     fn gen_clobber_save(
         call_conv: isa::CallConv,
         clobbers: &Set<Writable<RealReg>>,
+        fixed_frame_storage_size: u32,
     ) -> (u64, SmallVec<[Self::I; 16]>) {
         let mut insts = SmallVec::new();
         // Find all clobbered registers that are callee-save. These are only I64
         // registers (all XMM registers are caller-save) so we can compute the
         // total size of the needed stack space easily.
         let clobbered = get_callee_saves(&call_conv, clobbers);
-        let stack_size = 8 * clobbered.len() as u32;
+        let stack_size = 8 * clobbered.len() as u32 + fixed_frame_storage_size;
```

Comment:
Before this patch, `stack_size` was the same the clobbered size, which is the value that must be returned from this function. Now, it's the clobbered size + fixed frame size, so stack management is probably broken because of this (explaining random failures). Instead, only the clobbered size (which could have its own variable) should be returned.

Reply:
Thanks, I should have realised that.